### PR TITLE
update jax.lax.scan and test functions 

### LINKED ISSUE #21115: scan

### PR DESCRIPTION
**Pull request by Ahmed-abdelgalil**
Closes #21115

### CODE REVIEW

**Comment by Ahmed-abdelgalil** [comment]
hi @yopknopixx i just update my test scan function and run it locally and success i create new pull request because i rename branch from master to main can you please review my PR and kindly let me know if there is any Changes 

thanks!

File: ivy/data_classes/array/experimental/utility.py
Comment (yopknopixx) [inline ivy/data_classes/array/experimental/utility.py:1]: This is a Tax front-end task. Could you please explain why its necessary to modify this file?

File: ivy/data_classes/container/experimental/utility.py
Comment (yopknopixx) [inline ivy/data_classes/container/experimental/utility.py:1]: Same here

File: ivy/functional/ivy/experimental/utility.py
Comment (yopknopixx) [inline ivy/functional/ivy/experimental/utility.py:1]: And here

File: ivy_tests/test_ivy/test_functional/test_experimental/test_core/test_utility.py
Comment (yopknopixx) [inline ivy_tests/test_ivy/test_functional/test_experimental/test_core/test_utility.py:1]: And here.

**Review by yopknopixx — CHANGES_REQUESTED** [review]

**Comment by yopknopixx** [comment]
@Ahmed-abdelgalil I see some failing tests related to the Tax front-end. Could you please verify that your changes are not causing those failures?

File: ivy/data_classes/array/experimental/utility.py
Comment (Ahmed-abdelgalil) [reply ivy/data_classes/array/experimental/utility.py:1]: i didn't modify any one of them i just modify Control_flow_operatrors.py and test this modify done by root_yash

**Comment by Ahmed-abdelgalil** [comment]
@yopknopixx frontend failure at ivy_tests/test_ivy/test_frontends/test_jax/test_numpy/test_mathematical_functions.py::test_jax_polydiv,paddle

my changes at test_control_flow_operators.py::test_jax_scan don't cause failures

**Comment by Ahmed-abdelgalil** [comment]
@yopknopixx this happen while i try to sync fork and i was think that is done by one of engineers so i didn't modify them and modify only my task at control_flow_operators should i modify them delete changes at these files keep it ??

thanks

**Comment by yopknopixx** [comment]
@Ahmed-abdelgalil Please remove all changes not relevant to your task. Thanks

**Comment by yopknopixx** [comment]
> @yopknopixx frontend failure at ivy_tests/test_ivy/test_frontends/test_jax/test_numpy/test_mathematical_functions.py::test_jax_polydiv,paddle
> 
> my changes at test_control_flow_operators.py::test_jax_scan don't cause failures

Could you verify once that these tests are also failing on `unify:main` to be sure?

File: ivy/functional/ivy/experimental/utility.py
Comment (yopknopixx) [inline ivy/functional/ivy/experimental/utility.py:1]: It seems you deleted the contents of the file completely. Please undo your changes and keep the file as-was.

**Comment by yopknopixx** [comment]
@Ahmed-abdelgalil What we should have are only two modified files. `ivy/functional/frontends/jax/lax/control_flow_operators.py` and `ivy_tests/test_ivy/test_frontends/test_jax/test_lax/test_control_flow_operators.py`.
Could you try backing up your changes, doing a hard reset and then re-introducing your changes? That should clean things up.

**Comment by Ahmed-abdelgalil** [comment]
@yopknopixx I'm sorry for this confusion this my first time i hope every things clear now 

also there is new failure introduced at display test results what should i do with it 

thanks for your time

**Comment by yopknopixx** [comment]
@Ahmed-abdelgalil No problem, I'm happy to help. As for the new failure, you should make sure that your changes aren't the cause. To do that, you can check if the same tests are failing on `unifyai:main` without your changes, if they are, you're good to go!

**Pull request merged**